doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+                "BITS_0's value properly."
+        end if
+        call set0 % set(5)
+        if ( .not. set0 % none() ) then
+            write(*,*) "NONE interpreted SET0's value properly."
+        end if
+    end program demo_none
+```
+
+### `not` - Performs the logical complement on a bitset
+
+#### Status
+
+Experimental
+
+#### Description
+
+Performs the logical complement on the bits of `self`.
+
+#### Syntax
+
+`result = self % [[bitset_type(class):not(bound)]] ()`
+
+#### Class
+
+Elemental subroutine.
+
+#### Argument
+
+`self` shall be a scalar variable of class `bitset_type`. It is an
+  `intent(inout)` argument. On return its bits shall be the logical
+  complement of their values on input.
+
+#### Example
+
+```fortran
+    program demo_not
+        use stdlib_bitsets
+        type(bitset_large) :: set0
+        call set0 % init( 155 )
+        if ( set0 % none() ) then
+            write(*,*) "FROM_STRING interpreted " // &
+                "BITS_0's value properly."
+        end if
+        call set0 % not()
+        if ( set0 % all() ) then
+            write(*,*) "ALL interpreted SET0's value properly."
+        end if
+    end program demo_not
+```
+
+### `or` - Bitwise OR of the bits of two bitsets.
+
+#### Status
+
+Experimental
+
+#### Description
+
+Replaces the original bits of `set1` with the bitwise `or` of those
+bits with the bits of `set2`. Note `set1` and `set2` must have the
+samee number of bits, otherwise the result is undefined.
+
+#### Syntax
+
+`call [[stdlib_bitsets(module):or(interface)]](set1, set2)`
+
+#### Class
+
+Elemental subroutine.
+
+#### Arguments
+
+`set1`: shall be a scalar `bitset_64` or `bitset_large` variable. It
+ is an `intent(inout)` argument. On return the values of the bits in
+ `setf` are the bitwise `or` of the original bits in `set1` with the
+ corresponding bits in `set2`.
+
+`set2`: shall be a scalar expression of the same type as `set1`. It is
+an `intent(in)` argument. Note `bits(set2)` must equal `bits(set1)`
+otherwise the results are undefined.
+
+#### Example
+
+```fortran
+    program demo_or
+        use stdlib_bitsets
+        type(bitset_large) :: set0, set1
+        call set0 % init(166)
+        call set1 % init(166)
+        call or( set0, set1 ) ! none none
+        if ( none(set0) ) write(*,*) 'First test of OR worked.'
+        call set0 % not()
+        call or( set0, set1 ) ! all none
+        if ( all(set0) ) write(*,*) 'Second test of OR worked.'
+        call set0 % not()
+        call set1 % not()
+        call or( set0, set1 ) ! none all
+        if ( all(set0) ) write(*,*) 'Third test of OR worked.'
+        call set0 % not()
+        call or( set0, set1 ) ! all all
+        if ( all(set0) ) write(*,*) 'Fourth test of OR worked.'
+    end program demo_or
+```
+
+### `output` - Writes a binary representation of a bitset to a file
+
+#### Status
+
+Experimental
+
+#### Description
+
+Writes a binary representation of a bitset to an unformatted file.
```

Comment:
What kind of access is used? It might be important to mention it because the user must open the file himself.